tests/unit/parsers/test_collection.py

```diff
@@ -4,7 +4,43 @@
 from litezip.main import COLLECTION_NSMAP
 from lxml import etree
 
-from press.parsers import parse_collection_metadata
+from press.parsers import parse_collection_metadata, parse_collxml
+from press.models import CollectionElement
+
+
+def test_parse_collxml(collxml_templates):
+    with (collxml_templates / 'original.xml').open() as origin:
+        expected = parse_collxml(origin)
+
+    assert isinstance(expected, CollectionElement)
+    assert expected.name == 'collxml'
+
+
+def test_markup_in_title_gets_parsed(collxml_templates):
+    """Modules with markup in title gets parsed as part of the title.
+    """
+    titles_with_markup = [
+        'Introduction to Quartus and Circuit Diagram Design SOME MATH',
+        'Lab 1-1: 4-Bit Mux and all NAND/NOR Mux SOME STYLED TEXT',
+        'Lab 4-1 Interrupt Driven Programming in MSP430 Assembly '
+        'SOME DIV TAG SOME SPAN TAG',
+    ]
+
+    # TODO: ideally, the text that gets parsed should be the following
+    # (but it's tricky to do with a SAX parser):
```

Comment:
If this is really an issue you could look into lxml's parser, which I unfortunately may not have mentioned. https://lxml.de/parsing.html#iterparse-and-iterwalk It could help by allowing you to subscribe to specific elements and give you access to lxml.Element objects, which have an easier API to work with.

Reply:
That's interesting/useful. However, the problem isn't exactly with the SAX parser but with any stream parser, or more specifically, with trailing text using a stream parser.
This should hopefully demo the problem... for a title tag:
`<title>This is text <sub>and</sub>this is trailing text.</title>`

```
Start: This is text
Start: and
Start: this is trailing text.
```

Everything I tried gave me either this:
`This is text this is trailing text and`
Or this:
`and This is text this is trailing text`

It sort of occurred to me one way to parse the text correctly as part of the title tag is to be able to tell when I'm inside a title tag and any text after that point gets parsed as part of the tile but what would I do with the `sub` tag? Pretend it doesn't exist (ignore it completely and any others inside it)? Leave it alone but still pretend its text is actually directly within the `title` tag _somehow_?

Please, if you have any ideas, or any idea what I mean (I think I did a terrible job at explaining the problem)—let me know! @pumazi